functions/Get-DbaRepPublication.ps1

```diff
@@ -0,0 +1,137 @@
+
+    .EXAMPLE
+    Get-DbaRepPublication -SqlInstance sql2008, sqlserver2012
+    Return all publications for servers sql2008 and sqlserver2012.
+
+    .EXAMPLE
+    Get-DbaRepPublication -SqlInstance sql2008 -Database TestDB
+    Return all publications on server sql2008 for only the TestDB database
+
+    .EXAMPLE
+    Get-DbaRepPublication -SqlInstance sql2008 -PublicationType Transactional
+    Return all publications on server sql2008 for all databases that have Transactional publications
+#>
+    [CmdletBinding()]
+    Param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [DbaInstanceParameter[]]$SqlInstance, 
+        [object[]]$Database,
+        [PSCredential]$SqlCredential,
+        [ValidateSet("Transactional","Merge", "Snapshot")] 
+        [object[]]$PublicationType,
+        [switch]$EnableException
+    )
+
+    Begin{
+
+        if ($null -eq [System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SqlServer.RMO")) {
+            Write-Host "Replication management objects not available. Please install SQL Server Management Studio."
+        }
+
+    }
+
+    Process{
+
+        if (Test-FunctionInterrupt) { return }
+
+        foreach ($instance in $SqlInstance) {
+
+            # Connect to Publisher
+            try {
+                Write-Message -Level Verbose -Message "Connecting to $instance"
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential -MinimumVersion 9
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            $dbList = $server.Databases
+
+            if($Database) {
+                $dbList = $dbList | Where-Object name -in $Database
+            }
+
+            $dbList = $dbList | Where-Object { ($_.ID -gt 4) -and ($_.status -ne "Offline") }
+
+
+            foreach($db in $dbList) {
+
+                if(($db.ReplicationOptions -ne "Published") -and ($db.ReplicationOptions -ne "MergePublished")) {
+                    Write-Message -Level Verbose -Message "Skipping $($db.name). Database is not published."
+                }
+
+                $repDB = Connect-ReplicationDB -Server $server -Database $db
+                
+                $pubTypes = $repDB.TransPublications + $repDB.MergePublications
+
+                if($PublicationType) {
+                    $pubTypes = $pubTypes | Where-Object Type -in $PublicationType
+                }
+
+                foreach($pub in $pubTypes) {
+
+                    [PSCustomObject]@{
+                        Server = $server.name;
+                        Database = $db.name;
+                        PublicationName = $pub.Name;
+                        PublicationType = $pub.Type
+                    }      
+                }
+            }
+        }
+    }
+}
+
+#Helper Function
+function Connect-ReplicationDB {
```

Comment:
helper functions should be in `begin {}`

Reply:
Okay will do. I was needing to Mock this particular part of the code and I was unable to figure out how to mock it when the function was in begin {}. Any tips / examples? Thanks!

Reply:
Ah ha, I have no idea about mocking, perhaps @SQLDBAWithABeard has insight?

Reply:
I'll have a look tomorrow

Reply:
ty beard, @ColinDouglas92 - you can also add it to the internal/functions folder instead and it may work from there.